app/src/main/java/nerd/tuxmobil/fahrplan/congress/repositories/AppRepository.kt

```diff
@@ -448,23 +448,32 @@ object AppRepository {
         }
         val dayRanges = loadSessionsForAllDays(includeEngelsystemShifts = false)
                 .toDayRanges()
+        val oldShifts = loadEngelsystemShiftsForAllDays()
         val sessionizedShifts = shifts
                 .also { logging.d(LOG_TAG, "Shifts unfiltered = ${it.size}") }
                 .cropToDayRangesExtent(dayRanges)
                 .also { logging.d(LOG_TAG, "Shifts filtered = ${it.size}") }
                 .toSessionAppModels(logging, ENGELSYSTEM_ROOM_NAME, dayRanges)
                 .sanitize()
-        val toBeUpdatedSessions = loadSessionsForAllDays(false) // Drop all shifts before ...
+        val shiftChanges = computeSessionsWithChangeFlags(sessionizedShifts, oldShifts)
+        if (oldShifts.isEmpty() || shiftChanges.foundChanges) {
+            val toBeUpdatedSessions = loadSessionsForAllDays(false) // Drop all shifts before ...
                 .toMutableList()
-                // Shift rooms to make space for the Engelshifts room
-                .shiftRoomIndicesOfMainSchedule(sessionizedShifts.toDayIndices())
+                .let {
+                    when {
+                        // Shift rooms to make space for the Engelshifts room
+                        oldShifts.isEmpty() -> it.shiftRoomIndicesOfMainSchedule(sessionizedShifts.toDayIndices())
```

Comment:
Will this "forget" to shift sessions on days where you don't have angel shifts the first time you fetch angel shifts?

Wouldn't it be easier to always leave space for angel shifts?

Reply:
:thinking: Hm. This develops into its own discussion I feel.